### PR TITLE
GHOLD-388: Document restrictions when initializing interface fields in n4jsd

### PR DESCRIPTION
**Pull request by qtran-n4**
## Task
This PR documents the restrictions when initializing interface fields in n4jsd and is a complement to the PR done for https://github.com/NumberFour/n4js/issues/388

### CODE REVIEW

**Comment by qtran-n4** [comment]
@mor-n4 : Please have a look. I haven't used a matrix to describe this restriction because it is not worth it.

File: docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc [hunk 1]
```diff
@@ -786,6 +786,61 @@ new C( { s: "Hello", weird: true } );
 
 ====
 
+Restriction when initializing interface fields via @Spec constructor [[restriction-interface-field-spec-style-constructor]] ::
+
+In most cases, interface definitions in `n4jsd` files simply declare functions and fields that are supposed to by provided by the runtime environment.
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc]: typo: `supposed to by/be`

File: docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc [hunk 2]
```diff
@@ -786,6 +786,61 @@ new C( { s: "Hello", weird: true } );
...
+As a result, there are restrictions as to whether fields of interface defined in `n4jsd` files can initialized via `@Spec` constructors or not.
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc]: typo: `interfaceS`

File: docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc [hunk 3]
```diff
@@ -786,6 +786,61 @@ new C( { s: "Hello", weird: true } );
...
+}
+
+@N4JS
+export external interface J  {
+    public n: string;
+}
+----
+
+[source,n4js]
+----
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc]: add somewhere the file name of each listing and make sure it is easy to see where one file ends and the other starts.

File: docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc [hunk 4]
```diff
@@ -786,6 +786,61 @@ new C( { s: "Hello", weird: true } );
...
+export external interface I  {
+    public m: string;
+}
+
+@N4JS
+export external interface J  {
+    public n: string;
+}
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.spec/chapters/05_classifiers/classifiers-part-02-members.adoc]: you are not using interface `J` in the second file.
Either
* remove interface `J`, OR
* add a class `D` implementing `J` and correpsonding lines to show on the console that initialization actually works for field n in J.

**Comment by qtran-n4** [comment]
@mor-n4 Thanks for your review! Have integrated your comments in the new commits. Merging now.

**Pull request merged**